test/check/check.go

```diff
@@ -695,6 +695,50 @@ func checkForPrometheusRuleRemoval(name string) error {
 	return isNotFound("PrometheusRule", name, err)
 }
 
+func getMonitoringEndpoint() (*corev1.Endpoints, error) {
+	By("Finding CNAO prometheus endpoint")
+	endpoint := &corev1.Endpoints{}
+	err := framework.Global.Client.Get(context.Background(), types.NamespacedName{Name: MonitoringComponent.Service, Namespace: components.Namespace}, endpoint)
+	if err != nil {
+		return nil, err
+	}
+	return endpoint, nil
+}
+
+func scrapeEndpointAddress(epAddress corev1.EndpointAddress, epPort int32) (string, error) {
+	stdout, err := Kubectl("exec", "-n", epAddress.TargetRef.Namespace, epAddress.TargetRef.Name, "--", "curl", "-L", "-s", "-k", fmt.Sprintf("http://%s:%d/metrics", epAddress.IP, epPort))
```

Comment:
can you check please if curl 127.0.0.1:port works?
if so just the port will need to be fetched, and then maybe ep wont be need to be found
just use the port using the port geter from #928 if its suitable

Reply:
We already use the pod name in `epAddress.TargetRef.Name`, so why not use `epAddress.IP`?
Is there a reason we should't?

Reply:
I mean you totally actually don't need the ep at all
you can just use 127.0.0.1:<PORT> once you `kubectl exec` to the container.

Unless you do want to include the end point as part of the e2e test and then its ok,
basically the ep is not needed here just in order to fetch the metrics, since you are running from within the container.

Reply:
>Unless you do want to include the end point as part of the e2e test and then its ok

yep that's it.